### DISCUSSION TITLE
Ignore multipart requests

### DISCUSSION THREAD

**Discussion by bahbbc** [vcr/vcr#870] [Ideas]
I had a problem using VCR with multipart requests. The cassette keeps changing because the requests boundary was always changing. 

Everytime I ran the specs I had a new change in the cassette to add. To fix this we made this workaround:

``` ruby
VCR.configure do |config|
 # An integer (boundary) is inserted in some parts of body when request is multipart.
 # This matcher compares request bodies not considearing boundary
 config.register_request_matcher :body_without_boundary do |new_request, old_request|
   new_request_boundary = new_request.body.scan(/(?<=--)[0-9]*/)[0].to_s
   old_request_boundary = old_request.body.scan(/(?<=--)[0-9]*/)[0].to_s

   new_request.body.gsub(new_request_boundary, '') == old_request.body.gsub(old_request_boundary, '')
 end

 config.cassette_library_dir = 'spec/fixtures/vcr_cassettes'
 config.default_cassette_options[:match_requests_on] = [:method, :uri, :body_without_boundary]
 config.default_cassette_options[:record] = :new_episodes
 config.hook_into :webmock
end
```

**Comment by krainboltgreene** [comment]
Very interesting. The joys of HTTP request edge cases.

I'm surprised that vcr is making a proverbial checksum including body and not the verb + uri + version + headers.

**Comment by mcfiredrill** [comment]
Actually, it makes sense to me that you would need a custom matcher if the body is not the same.
Seems like expected behaviour, and not much we can do other than requiring that you write a custom matcher, unfortunately. 

Is this really a :bug: ?
